packages/jest-mock/src/index.ts

```diff
@@ -966,8 +1015,34 @@ export class ModuleMocker {
         ) {
           return;
         }
-        // @ts-expect-error no index signature
-        const slotMetadata = this.getMetadata<T>(component[slot], refs);
+
+        let descriptor = Object.getOwnPropertyDescriptor(component, slot);
+        let proto = Object.getPrototypeOf(component);
+        while (!descriptor && proto !== null) {
+          descriptor = Object.getOwnPropertyDescriptor(proto, slot);
+          proto = Object.getPrototypeOf(proto);
+        }
+
+        let slotMetadata: MockFunctionMetadata<T> | null = null;
```

Comment:
`MockFunctionMetadata` is renamed to `MockMetadata`. Both are still around, but better use `MockMetadata`:

```suggestion
        let slotMetadata: MockMetadata<T> | null = null;
```